main/emacs/APKBUILD

```diff
@@ -0,0 +1,186 @@
+arch="noarch"
+url="http://www.gnu.org/software/emacs/emacs.html"
+license="GPL3"
+depends="hicolor-icon-theme desktop-file-utils"
+makedepends="autoconf automake linux-headers paxmark
+	librsvg-dev giflib-dev libxpm-dev gtk+2.0-dev gconf-dev alsa-lib-dev
+	imagemagick-dev glib-dev fontconfig-dev libpng-dev
+	libxml2-dev pango-dev tiff-dev libjpeg-turbo-dev ncurses-dev
+	ncurses-libs gnutls-dev libxaw-dev"
+subpackages="$pkgname-doc $pkgname-nox $pkgname-x11 $pkgname-gtk2"
+source="ftp://ftp.gnu.org/gnu/emacs/emacs-$pkgver.tar.xz
+        gamedir.patch
+        noaslr-dump.patch
+        musl.patch"
+
+_builddir="$srcdir/emacs-$pkgver"
+prepare() {
+	local i
+	cd "$_builddir"
+	for i in $source; do
+		case $i in
+		*.patch) msg $i; patch -p1 -i "$srcdir"/$i || return 1;;
+		esac
+	done
+}
+
+build_x11() {
+	cd "$_builddir/x11"
+	./autogen.sh || return 1
+	CFLAGS=-fno-pie \
+	LDFLAGS=-no-pie \
+	./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--prefix=/usr \
+		--sysconfdir=/etc \
+		--libexecdir=/usr/lib \
+		--localstatedir=/var \
+		--without-makeinfo \
+		--with-gameuser=:games \
+      --with-x-toolkit=athena \
+      --without-toolkit-scroll-bars \
+		--without-dbus \
+		--with-xft \
+		--with-jpeg=yes \
+		--with-tiff=yes \
+		|| return 1
+	make
+}
+
+build_gtk2() {
+	cd "$_builddir/gtk2"
+	./autogen.sh || return 1
+	CFLAGS=-fno-pie \
+	LDFLAGS=-no-pie \
+	./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--prefix=/usr \
+		--sysconfdir=/etc \
+		--libexecdir=/usr/lib \
+		--localstatedir=/var \
+		--without-makeinfo \
+		--with-gameuser=:games \
+		--with-x-toolkit=gtk2 \
+		--with-xft \
+		--with-jpeg=yes \
+		--with-tiff=no \
+		|| return 1
+	make
+}
+
+build_nox() {
+	cd "$_builddir/nox"
+	./autogen.sh || return 1
+	CFLAGS=-fno-pie \
+	LDFLAGS=-no-pie \
+	./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--prefix=/usr \
+		--sysconfdir=/etc \
+		--libexecdir=/usr/lib \
+		--localstatedir=/var \
+		--without-makeinfo \
+		--with-gameuser=:games \
+		--without-sound \
+		--without-x \
+		--without-file-notification \
+		|| return 1
+	make
+}
+
+build() {
+	cd "$_builddir"
+	[[ -d nox ]] || {
+      mkdir -p nox
+	   mv * nox || true
+	   cp -a nox gtk2
+	   cp -a nox x11
+   }
+   build_nox
+   build_x11
+   build_gtk2
+}
+
+package() {
+   mkdir -p "$pkgdir" && return 0
+}
+
+doc() {
+   mkdir -p "$subpkgdir"
+	cd "$_builddir/gtk2"
+	make DESTDIR="$subpkgdir" install
+	# remove conflict with ctags package
+	mv "$subpkgdir"/usr/share/man/man1/ctags.1.gz "$subpkgdir"/usr/share/man/man1/ctags.emacs.1.gz
+   # only keep info and man directories, all other is in the specific package
+   rm -rf "$subpkgdir"/usr/bin
+   rm -rf "$subpkgdir"/usr/bin
+   rm -rf "$subpkgdir"/usr/lib
+   rm -rf "$subpkgdir"/usr/share/appdata
+   rm -rf "$subpkgdir"/usr/share/applications
+   rm -rf "$subpkgdir"/usr/share/emacs
+   rm -rf "$subpkgdir"/usr/share/icons
+   rm -rf "$subpkgdir"/var
```

Comment:
`rm` accepts one _or more_ paths, so you can (and should) give it all these paths at once instead of multiple calls.

```sh
rm -rf "$subpkgdir"/usr/bin \
	"$subpkgdir"/usr/bin \
	"$subpkgdir"/usr/lib \
	"$subpkgdir"/usr/share/appdata \
	"$subpkgdir"/usr/share/applications \
	"$subpkgdir"/usr/share/emacs \
	"$subpkgdir"/usr/share/icons \
	"$subpkgdir"/var
```